worldedit-core/src/main/java/com/sk89q/worldedit/extent/reorder/MultiStageReorder.java

```diff
@@ -94,58 +99,76 @@ public void setEnabled(boolean enabled) {
     }
 
     public boolean commitRequired() {
-        return stage1.size() > 0 || stage2.size() > 0 || stage3.size() > 0;
+        return enabled && stages.stream().anyMatch(stage -> stage.size() > 0);
+    }
+
+    /**
+     * Gets the stage priority of the block.
+     *
+     * @param block The block
+     * @return The priority
+     */
+    public int getPlacementPriority(BlockStateHolder block) {
```

Comment:
I think it might be nice to turn this into an enum, `PlacementPriority`:
```java
enum PlacementPriority {
  FIRST,
  LAST,
  LATE,
  FINAL
}
```

This should be a little more readable. Another option would be to just extract the numbers to fields, so there's no magic numbers around. It's hard to tell what `1` and `2` mean in `srcPriority == 1 || srcPriority == 2` below.